pkg/ingester/ingester.go

```diff
@@ -208,7 +208,7 @@ func New(cfg Config, clientConfig client.Config, store ChunkStore, limits *valid
 
 	if cfg.WAL.Enabled {
 		if err := os.MkdirAll(cfg.WAL.Dir, os.ModePerm); err != nil {
-			return nil, err
+			return nil, fmt.Errorf("error creating WAL folder at '%s': %w", cfg.WAL.Dir, err)
```

Comment:
Repeating `error` keyword in wrapped error message is often undesired, as it creates a lot of repetition both in code and for the reader when reported error is deeply nested.

I'd also suggest using `%q` instead of `'%s'`, which as far as I know is a recommended way of formatting quoted strings in Go.
```suggestion
      path := cfg.WAL.Dir
      
      // Best effort try to make path absolute for easier debugging.
      path, _ = filepath.Abs(cfg.WAL.Dir)
      if path == "" {
        path = cfg.WAL.Dir
      }
      
			return nil, fmt.Errorf("creating WAL folder at %q: %w", path, err)
```